### DISCUSSION TITLE
special assets properties from figma?

### DISCUSSION THREAD

**Discussion by pierroo** [dicebear/dicebear#372] [Q&A] [answered]
First, I apologize for the many topics/discussions I create here;
I hope they aren't out of place, but I see them as potential improvements of the library for those who may need more flexibility for their own avatar editor.

As of now when we go "the figma way", we can add color variation as part of an asset

But what if we want to add other specific properties / custom ones; like for example having some of the assets behind a "paywall" from our app?
Like some cosmetics may be "locked" for premium users etc?

Ideally, being able to add a "isLocked" property from the figma directly that would be exported and part of the schema file;
because I am afraid if I manually input such custom property in the schema file myself, I will have to redo it manually each time I re-build / re-export from figma my assets (with new / edited ones)

**Comment by FlorianKoerner** [comment, marked as answer]
These are features that should be built around avatar creation and not part of the library or avatar style itself.

Which options you offer in a free variant of your app and which options you offer in a paid variant is something you define before avatar creation. But this has nothing to do with the avatar creation itself.
